### PR TITLE
e2e node: fix -v support

### PR DESCRIPTION
**Pull request by pohly**
#### What type of PR is this?

/kind bug
/kind failing-test

#### What this PR does / why we need it:

Since 43539c855f59f3318de73602d091c62ee5863cda (first released in v1.30.0-alpha.2), the test/e2e/framework manages -v and -vmodule and uses them for a logger which writes to the Ginkgo output stream.

This did not work for test/e2e_node, because:
- logs.AddFlags(pflag.CommandLine) registers its own -v and -vmodule flags
- pflag.CommandLine.AddGoFlagSet(flag.CommandLine) skips the corresponding flags in the flag.CommandLine
- pflag.Parse() initializes the settings in the "logs" package even though those are not used at runtime

The solution is to not use the "logs" package.

#### Special notes for your reviewer:

Probably not worth trying to get into 1.30, but should get backported to it once released.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by ffromani** [comment]
/lgtm

the rationale fully makes sense to me

**Comment by pohly** [comment]
There's also the inverse problem: when the change in https://github.com/kubernetes/kubernetes/commit/43539c855f59f3318de73602d091c62ee5863cda successfully replaces the verbosity flags, klog verbosity no longer gets changed. That then has the effect that "traditional" klog calls use verbosity 0 even when -v is used.

The solution is to not just replace the values, but wrap them. See second commit.

@ffromani: can you take another look?

File: test/e2e/framework/ginkgologger.go [hunk 1]
```diff
@@ -49,14 +49,16 @@ var (
...
+	// consistently, we need to intercept the Set call. This
+	// can be done by swapping out the flag.Value for the -v and
+	// -vmodule flags with a wrapper which calls both.
 	var fs flag.FlagSet
 	logConfig.AddFlags(&fs)
 	fs.VisitAll(func(loggerFlag *flag.Flag) {
 		klogFlag := flag.CommandLine.Lookup(loggerFlag.Name)
 		if klogFlag != nil {
-			klogFlag.Value = loggerFlag.Value
+			klogFlag.Value = &valueChain{Value: loggerFlag.Value, parentValue: klogFlag.Value}
```
Comment (ffromani) [inline test/e2e/framework/ginkgologger.go:61]: nice!

**Review by ffromani — COMMENTED** [review]
/lgtm

also the new fix LGTM

**Review by endocrimes — APPROVED** [review]
/approve
/lgtm

**Comment by pohly** [comment]
Cherry-pick in https://github.com/kubernetes/kubernetes/pull/124415.

**Pull request merged**